### DISCUSSION TITLE
Feature: (embedded) Natural Resonance Theory (NRT) analysis for periodic system might be useful and attractive

### DISCUSSION THREAD

**Discussion by kirk0830** [deepmodeling/abacus-develop#4282] [Ideas]
### Background

In chemical-related community, people may care about the strength of chemical bond, which is a classical model to interpret the stability of substances and reaction thermodynamics and kinetics. Although bond order analysis like Mayer bond order analysis and several population analysis methods are well-developed, a decomposition of chemical bonds into "covalent component" and "ionic component" might still be helpful for analysis chemical bond strength with more details. Natural Resonance Theory (NRT) is a method projecting optimized wavefunction onto several Valence-Bond (VB)-like many-electron wavefunctions, normalize the projection, like decompose the optimized wavefunction into several interested resonance structure. An example would be, even if two chemical bonds have identical bond orders, they may still be different in bond composition, I mean how many percentage of covalent part and how many ionic part, this will be a reference to discuss about thermal stability of substances. But, all these are, unfortunately mainly in classical quantum chemistry field well-developed where isolated systems are main to investigate.   
Instead in periodic system, two aspects are worthwhile to talk about, the first one is support of periodic system wavefunction analysis by Multiwfn, the other is the condition of still "to-be-developed" wavefunction analysis methods. LCAO methods of constructing wavefunction provide good chances and easy mode for/of transferring those analysis methods to periodic system calculation.

Application example:
- https://link.springer.com/article/10.1007/s12274-015-0796-9

Reference:
- Glendening E D, Weinhold F. Natural resonance theory: I. General formalism[J]. Journal of computational chemistry, 1998, 19(6): 593-609.
- Glendening E D, Weinhold F. Natural resonance theory: II. Natural bond order and valency[J]. Journal of Computational Chemistry, 1998, 19(6): 610-627.
- Glendening E D, Badenhoop J K, Weinhold F. Natural resonance theory: III. Chemical applications[J]. Journal of computational chemistry, 1998, 19(6): 628-646.
- https://www.scm.com/doc/ADF/Input/Advanced_analysis.html#adfnbo-gennbo-nbo-analysis
- https://nbo6.chem.wisc.edu/tut_nrt.htm
- https://en.wikipedia.org/wiki/Natural_resonance_theory

### Describe the solution you'd like

Eventually more and more analysis methods should be supported because one single wavefunction solving software is far from enough, people care more about properties one software can calculate.

### Task list only for developers

- [ ] Notice possible changes of behavior
- [ ] Explain the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi

### Task list for Issue attackers (only for developers)

- [ ] Review and understand the proposed feature and its importance.
- [ ] Research on the existing solutions and relevant research articles/resources.
- [ ] Discuss with the team to evaluate the feasibility of implementing the feature.
- [ ] Create a design document outlining the proposed solution and implementation details.
- [ ] Get feedback from the team on the design document.
- [ ] Develop the feature following the agreed design.
- [ ] Write unit tests and integration tests for the feature.
- [ ] Update the documentation to include the new feature.
- [ ] Perform code review and address any issues.
- [ ] Merge the feature into the main branch.
- [ ] Monitor for any issues or bugs reported by users after the feature is released.
- [ ] Address any issues or bugs reported by users and continuously improve the feature.

**Comment by QuantumMisaka** [comment]
I do consider the feature is needed for analysis the bonding properties, which will be very useful in chemical and catalytic simulation problem.
